### PR TITLE
Writing query

### PR DESCRIPTION
**Pull request by jackdavidweber**
Finished week 5. Successful build passes all tests.

Got everything to work, but still need to make things look nicer and more efficient

### CODE REVIEW

File: walkthroughs/week-5-tdd/project/src/main/java/com/google/sps/FindMeetingQuery.java [hunk 1]
```diff
@@ -15,9 +15,126 @@
...
+import java.util.Arrays;
+import java.util.ArrayList; // import the ArrayList class
+import java.util.Set;
+
+
+public final class FindMeetingQuery {    
+  ArrayList<TimeRange> options;
+
+  /*Returns true if the time slot is a possible option with respect to duration*/
+  public boolean slotPossibleDuration(MeetingRequest request, int startSlot, int endSlot){
```
Comment (augerm) [inline walkthroughs/week-5-tdd/project/src/main/java/com/google/sps/FindMeetingQuery.java]: `slotPossibleDuration` --> `slotIsPossibleDuration`

File: walkthroughs/week-5-tdd/project/src/main/java/com/google/sps/FindMeetingQuery.java [hunk 2]
```diff
@@ -15,9 +15,126 @@
...
+      return (request.getDuration() <= (endSlot - startSlot));
+  }
+
+  public boolean conflictIsRelevant(Event conflict, Collection meetingAttendees){
+    Set<String> conflictAttendees = conflict.getAttendees();
+    // Collection<String> meetingAttendees = request.getAttendees();
```
Comment (augerm) [inline walkthroughs/week-5-tdd/project/src/main/java/com/google/sps/FindMeetingQuery.java]: Remove

File: walkthroughs/week-5-tdd/project/src/main/java/com/google/sps/FindMeetingQuery.java [hunk 3]
```diff
@@ -15,9 +15,126 @@
...
+
+    // returns true if the two sets have at least one attendee in common
+    return !(Collections.disjoint(conflictAttendees, meetingAttendees));
+  }
+
+  public boolean conflictIsOptionallyRelevant(Event conflict, MeetingRequest request){
```
Comment (augerm) [inline walkthroughs/week-5-tdd/project/src/main/java/com/google/sps/FindMeetingQuery.java]: `conflictIsOptionallyRelevant` --> `conflictIsRelevantForOptionalMembers`

File: walkthroughs/week-5-tdd/project/src/main/java/com/google/sps/FindMeetingQuery.java [hunk 4]
```diff
@@ -15,9 +15,126 @@
...
+    Collection<String> optionalAttendees = request.getOptionalAttendees();
+
+    // returns true if the two sets have at least one optional attendee in common
+    return !(Collections.disjoint(conflictAttendees, optionalAttendees));  }
 
-public final class FindMeetingQuery {
   public Collection<TimeRange> query(Collection<Event> events, MeetingRequest request) {
-    throw new UnsupportedOperationException("TODO: Implement this method.");
+    // Create collection of timeranges representing options for meeting
+    Collection<TimeRange> optionsCollection;
```
Comment (augerm) [inline walkthroughs/week-5-tdd/project/src/main/java/com/google/sps/FindMeetingQuery.java]: Variable name should mention something to do with timeRanges or availability. `options` is a bit ambiguous

File: walkthroughs/week-5-tdd/project/src/main/java/com/google/sps/FindMeetingQuery.java [hunk 5]
```diff
@@ -15,9 +15,126 @@
...
+    if (attendees.isEmpty() && optionalAttendees.isEmpty()){
+        optionsCollection = Arrays.asList(TimeRange.WHOLE_DAY);
+        return optionsCollection;
+    
+    // if there are optional attendees but no attendees, treat optional attendees as attendees
+    } else if (attendees.isEmpty()){
+        attendees = optionalAttendees;
+    }
+
+    Iterator<Event> eventsIterator =  events.iterator();
```
Comment (augerm) [inline walkthroughs/week-5-tdd/project/src/main/java/com/google/sps/FindMeetingQuery.java]: Do we need to sort this?

**Review by augerm — CHANGES_REQUESTED** [review]

File: walkthroughs/week-5-tdd/project/src/main/java/com/google/sps/FindMeetingQuery.java (hunk 5 above)
Comment (jackdavidweber) [reply walkthroughs/week-5-tdd/project/src/main/java/com/google/sps/FindMeetingQuery.java]: Yes- will work on this on Tuesday.

Comment (jackdavidweber) [reply walkthroughs/week-5-tdd/project/src/main/java/com/google/sps/FindMeetingQuery.java]: @augerm done!

**Review by augerm — APPROVED** [review]

**Pull request merged**